extensions/spring-xd-extension-batch/src/main/java/org/springframework/batch/step/tasklet/x/AbstractProcessBuilderTasklet.java

```diff
@@ -102,34 +163,37 @@ public RepeatStatus execute(StepContribution contribution, ChunkContext chunkCon
 			logger.error(msg);
 		}
 		finally {
-			printLog(commandName, commandLog, exitCode);
-			StringBuilder firstException = new StringBuilder();
-			if (exitCode != 0) {
-				for (String line : commandLog) {
-					if (firstException.length() == 0) {
-						if (line.contains("Exception")) {
-							firstException.append(line).append("\n");
-						}
-					}
-					else {
-						if (line.startsWith("\t")) {
-							firstException.append(line).append("\n");
-						}
-						else {
-							break;
-						}
-					}
+			if (p != null) {
+				p.destroy();
+			}
+			commandOut = getProcessOutput(out);
+			commandErr = getProcessOutput(err);
+			printLog(commandName, commandOut, commandErr);
```

Comment:
@ghillert and I were talking about this in a different capacity.  The `ExecutionContext` is intended for holding state of a job.  It's not intended to be holding entire log files which I'm sure we all agree can grow to be quite large.  This really goes to how logs in general should be aggregated in Spring XD.  Currently this is attempting to address one use case of log management, but there are many others that XD is starting to bump up against.  The use case I was talking with Gunnar about was that users want to see the full stack trace of an error that occurs in a batch job in the UI, however we don't persist it anywhere (we store the first 2500 characters of it in the job repository in most  situations).  I think we need to talk about a more general log aggregation strategy for Spring XD (that may just be punting to PCF) in general.